### PR TITLE
fix: add ConfigurationMetadata to ValueMetadata union

### PR DESCRIPTION
**Pull request by raman325**
`node.getValueMetadata` and `getCCValueMetadata` can return `ConfigurationMetadata` in addition to `ValueMetadata`

### CODE REVIEW

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: packages/zwave-js/src/lib/node/Node.ts [hunk 1]
```diff
@@ -605,7 +606,9 @@ export class ZWaveNode extends Endpoint {
 	 * Retrieves metadata for a given value id.
 	 * This can be used to enhance the user interface of an application
 	 */
-	public getValueMetadata(valueId: ValueID): ValueMetadata {
+	public getValueMetadata(
+		valueId: ValueID,
+	): ValueMetadata | ConfigurationMetadata {
```
[the remark this answers is not in the document]
Comment (raman325) [reply packages/zwave-js/src/lib/node/Node.ts]: should we move `ConfigValue` and `ValueFormat` as well?

[the remark this answers is not in the document]
Comment (raman325) [reply packages/zwave-js/src/lib/node/Node.ts]: What would you want the `type` to be?

[the remark this answers is not in the document]
Comment (raman325) [reply packages/zwave-js/src/lib/node/Node.ts]: Actually, thinking about it some more, I'm not sure that changing the type is a good idea. For every other type it describes the value type and can be used to figure out how to parse it. If we do want a unique type, I think we would want to name it something that makes it clear what data type the value is.

[the remark this answers is not in the document]
Comment (raman325) [reply packages/zwave-js/src/lib/node/Node.ts]: Well without this change, I would expect the type to be either `number`, `number[]`, or `boolean` in those cases no? So maybe we should just append `config_` as a prefix to those for config values?

[the remark this answers is not in the document]
Comment (raman325) [reply packages/zwave-js/src/lib/node/Node.ts]: or camel case so `configNumber` etc. Separately, are there any places where `number[]` is actually used? I found it defined in the code but haven't seen any use cases of it yet

[the remark this answers is not in the document]
Comment (AlCalzone) [reply packages/zwave-js/src/lib/node/Node.ts]: Let's keep it like it is for now - if it seems necessary, we can add a type to distinguish.

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**